android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -189,25 +179,94 @@ public void sendToNotificationCentre(Bundle bundle) {
             PendingIntent pendingIntent = PendingIntent.getActivity(context, notificationID, intent,
                     PendingIntent.FLAG_UPDATE_CURRENT);
 
-            notificationBuilder.setContentIntent(pendingIntent);
+            NotificationCompat.Builder summaryBuilder = new NotificationCompat.Builder(context,
+                    NOTIFICATION_CHANNEL_ID)
+                    .setSmallIcon(smallIconResId)
+                    .setStyle(new NotificationCompat.InboxStyle().setSummaryText(APP_ROOT_NAME))
+                    .setGroup(APP_BUNDLE_ID).setGroupSummary(true)
+                    .setAutoCancel(bundle.getBoolean("autoCancel", true));
+
+            String sender = bundle.getString("sender");
+            String chatMessage = bundle.getString("chat_message");
+            String chatTimestamp = bundle.getString("chat_timestamp");
+
+            if (
+                    bundleTitle != null &&
+                    bundleId != null &&
+                    notificationTypeInt == RB_GROUP_MSG_TYPE &&
+                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                    sender != null &&
+                    chatMessage != null &&
+                    chatTimestamp != null
+            ) {
 
-            if (bundle_title != null && bundle_id != null) {
                 // LP: is supposed to be grouped message
-                int bundle_id_int = Integer.parseInt(bundle_id);
+                int bundleIdInt = Integer.parseInt(bundleId);
 
-                notificationBuilder.setGroup(bundle_id);
+                Bundle extras = new Bundle();
 
-                NotificationCompat.Builder summaryBuilder = new NotificationCompat.Builder(context,
-                        NOTIFICATION_CHANNEL_ID).setSmallIcon(smallIconResId)
-                                .setStyle(new NotificationCompat.InboxStyle().setSummaryText(bundle_title))
-                                .setGroup(bundle_id).setGroupSummary(true);
-                notificationManager.notify(notificationID, notificationBuilder.build());
-                notificationManager.notify(bundle_id_int, summaryBuilder.build());
+                for (StatusBarNotification notif : notificationManager.getActiveNotifications()) {
+                    if (notif.getId() == bundleIdInt ) {
+                        extras = notif.getNotification().extras;
+                    }
+                }
+
+                ArrayList<String> existingUsernames = extras.getStringArrayList("existingUsernames");
+                if (existingUsernames == null) {
+                    existingUsernames = new ArrayList<>();
+                }
+                existingUsernames.add(sender);
+                extras.putStringArrayList("existingUsernames", existingUsernames);
```

Comment:
can you extract "existingUsernames", "existingMessages" and "existingTImestamps" to contstants called `EXTRAS_KEY_USERNAMES`, etc. ?

Reply:
Sure